src/views/Contacts.vue

```diff
@@ -313,6 +322,57 @@ export default {
 			}
 		},
 
+		/**
+		 * Download group of contacts
+		 *
+		 * @param {Object} group of contacts to be downloaded
+		 * @returns {Function} trigger download of resulting single vcard file
+		 */
+		downloadGroup(group) {
+			const contacts = this.contacts
+			const contactIds = group.contacts
+			// find addressbooks and vcard urls
+			const urls = contactIds.map(cid => ({
+				addressbook: contacts[cid].addressbook.id,
+				url: _.last(contacts[cid].url.split('/'))
+			}))
+			// group vcard urls by addressbook
+			const groupedUrls = _.mapValues(
+				_.groupBy(urls, 'addressbook'),
+				group => group.map(contact => contact.url)
+			)
+			// create vcard promise with the requested contacts
+			const vcardPromise = client.addressBookHomes[0]				// question: is that generic enough?
+				// promise #1: get address books
+				.findAllAddressBooks()
+				// promise #2: multiget for every address book
+				.then(addressbooks => {
+					const pairs = _.toPairs(groupedUrls)
+					const responses = pairs
+						.map(pair => {
+							const addressbook = addressbooks.find(ab => ab.displayname === pair[0])
+							return addressbook.addressbookMultigetExport(pair[1])
+						})
+					return Promise.all(responses)
+				})
+				// promise #3: merge into one object
+				.then(response => {
+					return response.map(data => data.body).join('')	// question: still to much heavy lifting in the browser?
+				})
+			// download vcard
+			return function() {
```

Comment:
Maybe create a dedicated function to download the data as a file, for readability? :)